search-service/src/test/kotlin/com/egm/stellio/search/service/EntityAccessRightsServiceTests.kt

```diff
@@ -262,4 +267,100 @@ class EntityAccessRightsServiceTests : WithTimescaleContainer {
+    fun `it should allow user who have read right on entity`() {
+        val specificAccessPolicies =
+            listOf(AuthContextModel.SpecificAccessPolicy.AUTH_READ, AuthContextModel.SpecificAccessPolicy.AUTH_WRITE)
+
+        entityAccessRightsService.setReadRoleOnEntity(subjectUuid, entityId).block()
+
+        every { temporalEntityAttributeService.hasSpecificAccessPolicies(any(), any()) } returns Mono.just(false)
+
+        runBlocking {
+            entityAccessRightsService.hasReadRightOnEntity(Some(subjectUuid), entityId, specificAccessPolicies).fold(
+                { Assertions.assertTrue(it == null) },
+                { Assertions.assertTrue(it != null) }
+            )
+        }
+    }
+
+    @Test
+    fun `it should allow user who have write right on entity`() {
+        val specificAccessPolicies =
+            listOf(AuthContextModel.SpecificAccessPolicy.AUTH_READ, AuthContextModel.SpecificAccessPolicy.AUTH_WRITE)
+
+        entityAccessRightsService.setWriteRoleOnEntity(subjectUuid, entityId).block()
+
+        every { temporalEntityAttributeService.hasSpecificAccessPolicies(any(), any()) } returns Mono.just(false)
+
+        runBlocking {
+            entityAccessRightsService.hasWriteRightOnEntity(Some(subjectUuid), entityId, specificAccessPolicies).fold(
+                { Assertions.assertTrue(it == null) },
+                { Assertions.assertTrue(it != null) }
+            )
+        }
+    }
+    @Test
+    fun `it should not allow user who have not read right on entity or specific access policy`() {
+        val specificAccessPolicies =
+            listOf(AuthContextModel.SpecificAccessPolicy.AUTH_READ, AuthContextModel.SpecificAccessPolicy.AUTH_WRITE)
+
+        every { temporalEntityAttributeService.hasSpecificAccessPolicies(any(), any()) } returns Mono.just(false)
+
+        runBlocking {
+            entityAccessRightsService.hasReadRightOnEntity(Some(subjectUuid), entityId, specificAccessPolicies).fold(
+                { assertEquals("User forbidden read access to entity $entityId", it.message) },
+                { Assertions.assertTrue(it == null) }
+            )
+        }
+    }
+
+    @Test
+    fun `it should not allow user who have not write right on entity or specific access policy`() {
+        val specificAccessPolicies =
+            listOf(AuthContextModel.SpecificAccessPolicy.AUTH_READ, AuthContextModel.SpecificAccessPolicy.AUTH_WRITE)
+
+        every { temporalEntityAttributeService.hasSpecificAccessPolicies(any(), any()) } returns Mono.just(false)
+
+        runBlocking {
+            entityAccessRightsService.hasWriteRightOnEntity(Some(subjectUuid), entityId, specificAccessPolicies).fold(
+                { assertEquals("User forbidden write access to entity $entityId", it.message) },
+                { Assertions.assertTrue(it == null) }
+            )
+        }
+    }
+    @Test
+    fun `it should allow user who have specific access policy without test read right on entity`() {
```

Comment:
the user does not have a specific access policy (but the target entity may have)

Reply:
don't really get the point of this test and the following one

Reply:
I test that if there is an access policy on the entity it returns me that the user can do things on this one